### PR TITLE
fix: Properly parse date strings

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Properly parse the new date format, which contains a time offset.

### CODE REVIEW

**Comment by Isira-Seneviratne** [comment]
> today when i created a branch to work on the instance button issue, i merged this and app is crashing with same datetimeparse errors, but the current alpha with new dates is working fine!

What's the date format displayed in the error message?

**Comment by Bnyro** [comment]
We currently still only use the date without time. Wouldn't it make more sense to use `Instant` for everything (so that hours, minutes and seconds are taking into consideration too) and parse `LocalDate` instances to an `Instant` at `00:00` instead?

**Comment by Isira-Seneviratne** [comment]
> We currently still only use the date without time. Wouldn't it make more sense to use `Instant` for everything (so that hours, minutes and seconds are taking into consideration too) and parse `LocalDate` instances to an `Instant` at `00:00` instead?

Yeah, it would.

File: app/src/main/java/com/github/libretube/db/obj/WatchHistoryItem.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ import kotlinx.serialization.Serializable
 
 @Serializable
 @Entity(tableName = "watchHistoryItem")
-data class WatchHistoryItem(
+data class WatchHistoryItem constructor(
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/db/obj/WatchHistoryItem.kt:13]: What is this change needed for?

**Review by Bnyro — APPROVED** [review]

File: app/src/main/java/com/github/libretube/db/obj/WatchHistoryItem.kt (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/com/github/libretube/db/obj/WatchHistoryItem.kt:13]: I accidentally committed that change.
